test/validation/api/timer/timer.c

```diff
@@ -488,6 +511,23 @@ static void timer_test_event_type(odp_queue_type_t queue_type,
 	int num = 5;
 	odp_timer_t timer[num];
 
+	memset(&capa, 0, sizeof(capa));
+	ret = odp_timer_capability(ODP_CLOCK_CPU, &capa);
+	CU_ASSERT_FATAL(ret == 0);
+
+	if (!(capa.queue_type_plain && capa.queue_type_sched)) {
```

Comment:
Test cases should be modified to be conditional based on capability. If a test case returns without testing anything, it still looks like a passed validation test case.

You can take example from other suites how ODP_TEST_INFO_CONDITIONAL() macro is used. Basically, e.g. ODP_TEST_INFO(timer_test_pkt_event_sched) would change to ODP_TEST_INFO_CONDITIONAL(timer_test_pkt_event_sched, check_sched_queue_support). Suite init phase would record capabilities, and a check function would read those and return ACTIVE / INACTIVE status.

For example,
```
static int check_sched_queue_support(void)
{
	if (globals->capa.queue_type_sched)
		return ODP_TEST_ACTIVE;

	return ODP_TEST_INACTIVE;
}
```

Reply:
Thanks for the pointers, I will modify it in the next version.